src/components/Main/Schedule/services.ts

```diff
@@ -0,0 +1,172 @@
+    const day = jsDate.getDate();
+    return {
+      Header: zeroingDates(month, day),
+      accessor: zeroingDates(month, day),
+    };
+  });
+}
+
+export function isInTime(target: string, start: string, end: string): boolean {
+  const targetTime = timeToMilliSeconds(target);
+  const startTime = timeToMilliSeconds(start);
+  const endTime = timeToMilliSeconds(end);
+  return startTime <= targetTime && targetTime <= endTime;
+}
+
+export function timeToMilliSeconds(time: string) {
+  const [hourMin, period] = time.split(' ');
+  const [hour, min] = hourMin.split(':');
+  let convertedHour = hour === '12' ? 3600000 : 3600000 * parseInt(hour);
+  let convertedMin = 60000 * parseInt(min);
+  let convertedPeriod = period === 'AM' ? 0 : 43200000;
+
+  return convertedHour + convertedMin + convertedPeriod;
+}
+
+export function createColumns(daterange: any, day: any) {
+  return [
+    {
+      Header: 'Time',
+      accessor: 'time',
+    },
+    {
+      Header: 'Duty',
+      accessor: 'duty',
+    },
+    ...columnizedDates(everyRepeatingDayBetweenTwoDates(daterange[0], daterange[1], day)),
+  ];
+}
+
+export const contrivedDate = (date: string) => {
+  const jsDate = new Date(date);
+  const month = jsDate.getMonth();
+  const day = jsDate.getDate();
+  return zeroingDates(month, day);
+};
+
+export const makeData = (value: number): MappedScheduleInterface[] => {
+  const { daterange, weeklyEvents } = SCHEDULE[value];
+  let allSchedulesForTheWeek: MappedScheduleInterface[] = [];
+  weeklyEvents.map((schedule: WeeklyEventData, index: number) => {
+    const { day, events, dividers } = weeklyEvents[index];
+    const columns = createColumns(daterange, day);
+    let fullDaySchedule: MappedScheduleInterface[] = [];
+
+    dividers.map((divider: Divider) => {
+      let data: WeeklyAssignmentInterface[] = [];
+      let { name, timerange } = divider;
+
+      events.map((event: EventData) => {
+        let { duties, time } = event;
+        let everyWeeksAssignment: WeeklyAssignmentInterface[] = [];
+        if (isInTime(time, timerange.start, timerange.end))
+          duties.map((duty: DutyData, index: number) => {
+            const { title, tasks } = duty;
+            let assignments: WeeklyAssignmentInterface = { duty: title };
+            if (index === 0) assignments.time = time;
+
+            tasks.map((task: TaskData) => {
+              const { date, assignee } = task;
+              const assignedDate = contrivedDate(date);
+              assignments[assignedDate] = assignee;
+            });
+
+            everyWeeksAssignment.push(assignments);
+          });
+        data = [...data, ...everyWeeksAssignment];
+      });
+
+      fullDaySchedule.push({ day, columns, data, name });
+    });
+
+    allSchedulesForTheWeek = [...allSchedulesForTheWeek, ...fullDaySchedule];
+  });
+  return allSchedulesForTheWeek;
+};
+
+export const memoizeData = (data: any) => {
+  const makeDataLevel = () => data;
+  return makeDataLevel();
+};
```

Comment:
right now `memoizeData` looks like dead code (unused).

Reply:
It is, but technically react-table is required to have memoized data.. I'm gonna keep this in for now